src/it/resources/com/google/checkstyle/test/chapter6programpractice/rule62donotignoreexceptions/InputEmptyFinallyBlock.java

```diff
@@ -2,90 +2,108 @@
 
 class InputEmptyFinallyBlock {
+  boolean flag;
+
+  void doSm() {}
+
+  void foo() {
+    try {
+      if (!flag) {
+        doSm();
+      }
+    } catch (Exception e) {
+      /* ignore */
+    } finally { // ok
+      /* ignore */
+    } // ok
+  }
+
+  void foo2() {
+    try {
+      if (!flag) {
+        doSm();
+      }
+    } catch (Exception e) {
+      // do nothing
+    } finally { } // violation 'Empty finally block.'
+  }
+
+  class Inner {
     boolean flag;
+
     void doSm() {}
+
     void foo() {
-        try {
-           if (!flag) {
-               doSm();
-           }
-        } catch (Exception e) { /* ignore */ } //ok
-        finally {/* ignore */} //ok
+      try {
+        if (!flag) {
+          doSm();
+        }
+      } catch (Exception e) {
+        /* ignore */
+      } finally { // ok
+        /* ignore */
+      } // ok
     }
 
     void foo2() {
-        try {
-           if (!flag) {
-               doSm();
-           }
-        } catch (Exception e) {}
-        finally {} // violation 'Empty finally block.'
+      try {
+        if (!flag) {
+          doSm();
+        }
+      } catch (Exception e) {
+        // do nothing
+      } finally { } // violation 'Empty finally block.'
     }
+  }
 
-    class Inner {
-        boolean flag;
-        void doSm() {}
-        void foo() {
-            try {
-               if (!flag) {
-                   doSm();
-               }
-            } catch (Exception e) { /* ignore */ } //ok
-            finally {/* ignore */} //ok
-        }
+  Inner anon =
+          new Inner() {
+            boolean flag;
 
-        void foo2() {
-            try {
-               if (!flag) {
-                   doSm();
-               }
-            } catch (Exception e) {}
-            finally {} // violation 'Empty finally block.'
-        }
-    }
+            void doSm() {}
 
-    Inner anon = new Inner(){
-        boolean flag;
-        void doSm() {}
-        void foo() {
-            try {
-               if (!flag) {
-                   doSm();
-               }
-            } catch (Exception e) { /* ignore */ } //ok
-            finally {/* ignore */} //ok
-        }
+            void foo() {
+              try {
+                if (!flag) {
+                  doSm();
+                }
+              } catch (Exception e) {
+                /* ignore */
+              } finally { // ok
+                /* ignore */
+              } // ok
+            }
 
-        void foo2() {
-            try {
-               if (!flag) {
-                   doSm();
-               }
-            } catch (Exception e) {}
-            finally {} // violation 'Empty finally block.'
-        }
-    };
+            void foo2() {
+              try {
+                if (!flag) {
+                  doSm();
+                }
+              } catch (Exception e) {
+                // do nothing
+              } finally { } // violation 'Empty finally block.'
+            }
+          };
```

Comment:
@Zopsss , did you add `// do nothing` comment in catch ?

Reply:
Yes otherwise verifyWithWholeConfig would give error for `empty catch block`

Reply:
Should I remove it?

Reply:
hmmm, Does it mean that we have defect to detect the make the same violation on `{}` ?

Reply:
![Screenshot from 2024-07-13 07-23-05](https://github.com/user-attachments/assets/1a0f15ec-008a-47ee-9f17-5d6c03d3b447)

Reply:
This is https://checkstyle.org/google_style.html#a6.2

> [EmptyBlock](https://checkstyle.org/checks/blocks/emptyblock.html#EmptyBlock) with option=text.

Reply:
`{}`
this should be violation https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s6.2-caught-exceptions
please restore `{}` and lets see why we did not have violation before.

Reply:
>Yes otherwise verifyWithWholeConfig would give error for empty catch block

we are not migrated yet to WholeConfig 
https://github.com/checkstyle/checkstyle/blob/3c17a903d2a0c5abf303e1ae0c36458e2e89d224/src/it/java/com/google/checkstyle/test/chapter6programpractice/rule62donotignoreexceptions/CaughtExceptionsNotIgnoredTest.java#L36

ant it is violation from same Check.
We just need to do formatting, any changes in code with new elements is red flag.

Reply:
here is a problem:
```
rivanov@p5510:/var/tmp$ cat Test.java 
class InputEmptyFinallyBlock {
    boolean flag;
    void doSm() {}
    void foo() {
        try {
           if (!flag) {
               doSm();
           }
        } catch (Exception e) { /* ignore */ } //ok
        finally {/* ignore */} //ok
    }

    void foo2() {
        try {
           if (!flag) {
               doSm();
           }
        } catch (Exception e) {}
        finally {} // violation 'Empty finally block.'
    }
}rivanov@p5510:/var/tmp$ java -jar checkstyle-10.17.0-all.jar -c /google_checks.xml Test.java | grep Empty
[WARN] /var/tmp/Test.java:3:5: 'METHOD_DEF' should be separated from previous line. [EmptyLineSeparator]
[WARN] /var/tmp/Test.java:4:5: 'METHOD_DEF' should be separated from previous line. [EmptyLineSeparator]
[WARN] /var/tmp/Test.java:10:17: WhitespaceAround: '{' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] /var/tmp/Test.java:18:31: Empty catch block. [EmptyCatchBlock]
[WARN] /var/tmp/Test.java:18:31: WhitespaceAround: '{' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] /var/tmp/Test.java:19:17: Empty finally block. [EmptyBlock]
[WARN] /var/tmp/Test.java:19:17: WhitespaceAround: '{' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]

```

there is EmptyCatchBlock check , but we have in table and in chapter EmptyBlock

Reply:
Please do:
create new issue to put EmptyCatchBlock to coverage of chapter.
update this PR to keep `{}`. Until "Whole" is applied it will be without violation.
We will merge this PR sooner and in next PR, for migration to Whole, violation will appear.

Reply:
> create new issue to put EmptyCatchBlock to coverage of chapter.

This module has been already used by [4.1.3 Empty blocks: may be concise](https://checkstyle.sourceforge.io/styleguides/google-java-style-20180523/javaguide.html#s4.1.3-braces-empty-blocks), this section will cover our requirement, I guess there is no need to update the coverage page.

Reply:
Chapter clearly state no empty catch blocks, we should add, even it is a reference to the same instance of this Check.

Reply:
Should I revert 

```java
              } catch (Exception e) {
                // do nothing
              } finally { } // violation 'Empty finally block.'
```

to 

```java
            } catch (Exception e) {}
            finally {} // violation 'Empty finally block.'
```

When we will migrate to whole config, I will add violation for the EmptyCatchBlock

Reply:
> hmmm, Does it mean that we have defect to detect the make the same violation on {} ?

There's no defect in our implementation, we're able to detect the empty catch block, there's violation for it as you can see in your comment: https://github.com/checkstyle/checkstyle/pull/15248#discussion_r1676836582

> [WARN] /var/tmp/Test.java:18:31: Empty catch block. [EmptyCatchBlock]

We detect this violation as part of section [4.1.3 Empty blocks: may be concise](https://checkstyle.sourceforge.io/styleguides/google-java-style-20180523/javaguide.html#s4.1.3-braces-empty-blocks)

Reply:
> Chapter clearly state no empty catch blocks, we should add, even it is a reference to the same instance of this Check.
> 
> 6.2 should have it in table. No extra instance of this Check in config.

Ignore my above replies, I didnt saw your comment before I sent those replies.

Reply:
No defects in google_checks.xml, but there is in coverage table, violation comment should appear by migration to whole config usage. 
But we still should forget to update table , so we need extra issue as reminder to do more minor update.

Reply:
Issue opened: https://github.com/checkstyle/checkstyle/issues/15252

Reply:
Ok, remove extra comments From input to let this PR be approved.

Reply:
done, reverted those changes:

https://github.com/checkstyle/checkstyle/pull/15248/files#diff-c9f8591e129f37b79428658cb59370231680afda5bbcc93fa261384f1e13193fL66-R87